lib/Connection.php

```diff
@@ -67,14 +79,52 @@ public function __construct(string $uri)
         });
     }
 
+    /**
+     * When using the "redis" schemes the URI is parsed according
+     * to the rules defined by the provisional registration documents approved
+     * by IANA. If the URI has a password in its "user-information" part or a
+     * database number in the "path" part these values override the values of
+     * "password" and "database" if they are present in the "query" part.
```

Comment:
These are not well-defined. I think we should error if `password` or `database` is present in path and query.